CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/SyntaxRewriterFile.swift

```diff
@@ -30,7 +30,19 @@ let syntaxRewriterFile = SourceFileSyntax(leadingTrivia: generateCopyrightHeader
     open class SyntaxRewriter
     """
   ) {
-    DeclSyntax("public init() {}")
+    DeclSyntax(
+      """
+      let arena: SyntaxArena?
+      """
+    )
+
+    DeclSyntax(
+      """
+      public init(arena: SyntaxArena? = nil) {
```

Comment:
Actually, I think it would be better if all nodes created by a `SyntaxRewriter` live in the same `SyntaxArena`. Creating a `SyntaxArena` is fairly costly because it creates a new bump pointer allocator and it’s a waste to create a new one for every node. So, what I’m saying is that `arena` can be non-optional and this could be

```suggestion
      public init(arena: SyntaxArena = SyntaxArena()) {
```

Reply:
I tried that, but unfortunately the tests are failing. The `MacroApplication` SyntaxRewriter crashes internally.